tools/font-subset/test.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+#
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+'''
+Tests for font-subset
+'''
+
+import filecmp
+import os
+import subprocess
+import sys
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+SRC_DIR = os.path.normpath(os.path.join(SCRIPT_DIR, '..', '..', '..'))
+MATERIAL_TTF = os.path.join(SCRIPT_DIR, 'fixtures', 'MaterialIcons-Regular.ttf')
+IS_WINDOWS = sys.platform.startswith(('cygwin', 'win'))
+EXE = '.exe' if IS_WINDOWS else ''
+BAT = '.bat' if IS_WINDOWS else ''
+FONT_SUBSET = os.path.join(SRC_DIR, 'out', 'host_debug', 'font-subset' + EXE)
+if not os.path.isfile(FONT_SUBSET):
+  FONT_SUBSET = os.path.join(SRC_DIR, 'out', 'host_debug_unopt', 'font-subset' + EXE)
+if not os.path.isfile(FONT_SUBSET):
+  raise Exception('Could not locate font-subset%s in host_debug or host_debug_unopt - build before running this script.' % EXE)
+
+COMPARE_TESTS = (
+  (True,  '1.ttf', MATERIAL_TTF, [r'57347']),
+  (True,  '1.ttf', MATERIAL_TTF, [r'0xE003']),
+  (True,  '1.ttf', MATERIAL_TTF, [r'\uE003']),
+  (False, '1.ttf', MATERIAL_TTF, [r'57348']), # False because different codepoint
+  (True,  '2.ttf', MATERIAL_TTF, [r'0xE003', r'0xE004']),
+  (True,  '2.ttf', MATERIAL_TTF, [r'0xE003', r'0xE004', r'57347',]), # Duplicated codepoint
+  (True,  '3.ttf', MATERIAL_TTF, [r'0xE003', r'0xE004', r'0xE021',]),
+)
+
+FAIL_TESTS = [
+  ([FONT_SUBSET, 'output.ttf', 'does-not-exist.ttf'], ['1',]), # non-existant input font
```

Comment:
How about an empty string for a codepoint.

Reply:
Updated to add that case and to make sure it fails.

Reply:
That isn't empty, it's a space ;-)